test/functional/userAuth.test.js

```diff
@@ -40,6 +45,40 @@ describe('Authentication', () => {
     })
   })
 
+  describe('when logging in', () => {
+    before(() => sinon.stub(process.env, 'APPROVED_DOMAINS').value('(.* )?st.com'))
```

Comment:
The value of `process.env.APPROVED_DOMAINS` is read in at server initialization, which means stubbing it at this point in the test won't have any effect on authentication behavior. Instead, we could modify the `process.env.APPROVED_DOMAINS` value set in `bootstrap.js` to include a few different patterns at the same time: a single email, a regex, and a domain, then test each of those individually in this file.